sklearn/metrics/scorer.py

```diff
@@ -272,14 +274,122 @@ def check_scoring(estimator, scoring=None, allow_none=False):
                              'Please use `make_scorer` to convert a metric '
                              'to a scorer.' % scoring)
         return get_scorer(scoring)
-    elif hasattr(estimator, 'score'):
-        return _passthrough_scorer
-    elif allow_none:
-        return None
+    elif scoring is None:
+        if hasattr(estimator, 'score'):
+            return _passthrough_scorer
+        elif allow_none:
+            return None
+        else:
+            raise TypeError(
+                "If no scoring is specified, the estimator passed should "
+                "have a 'score' method. The estimator %r does not."
+                % estimator)
+    else:
+        raise ValueError("scoring value should either be a callable, string or"
+                         " None. %r was passed" % scoring)
+
+
+def _check_multimetric_scoring(estimator, scoring=None, allow_none=False):
+    """Check the scoring parameter in cases when multiple metrics are allowed
+
+    Parameters
+    ----------
+    estimator : sklearn estimator instance
+        The estimator for which the scoring will be applied.
+
+    scoring : string, callable or None, optional, default: None
+        A single string (see :ref:`_scoring_parameter`) or a callable
+        (see :ref:`_scoring`) to evaluate the predictions on the test set.
+
+        For evaluating multiple metrics either give a list of unique strings
+        or a dict with names as keys and callables as values.
+
+        NOTE that when using custom scorers, each scorer should return a single
+        value. Single scorers returning a list/array of values may be wrapped
+        into multiple scorers that return one value each.
+
+        If ``None``, the estimator's default scorer, is used.
+        The return value in that case will be ``{'score': <default_scorer>}``.
+        If the estimator's default scorer is not available, a ``TypeError``
+        is raised.
+
+    allow_none : boolean, default: False
+        Whether to allow None even if estimator does not have a score method.
+
+    Returns
+    -------
+    scorers_dict : dict
+        A dict mapping each scorer name to its validated scorer.
+
+        The validated scorer could be None if ``allow_none`` is ``True``
+        (``{'score': None}``).
+
+        The scorer_name is set to ``'score'`` if the default scorer is
```

Comment:
I don't understand that comment. What is ``scorer_name``? the key in the scoring dictionary? This should be in the user guide then.